### ISSUE TITLE
HSM support in v0.10.0-rc2 doesn't compile on FreeBSD 13.1.

### ISSUE THREAD

**Issue by ximon18** [NLnetLabs/krill#861]
Thanks to @Jakker for reporting that compilation fails on FreeBSD. I have reproduced this issue on an AWS EC2 VM running FreeBSD 13.1:

```
error: failed to run custom build command for `cryptoki-sys v0.1.3`

Caused by:
  process didn't exit successfully: `/tmp/cargo-install0pBOgH/release/build/cryptoki-sys-75d745693ea20a15/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'Compilation target (architecture, OS) tuple (x86_64, freebsd) is not part of the supported tuples. Please compile with the "generate-bindings" feature or add support for your platform :)', /home/ec2-user/.cargo/registry/src/github.com-1ecc6299db9ec823/cryptoki-sys-0.1.3/build.rs:26:17
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to compile `krill v0.10.0-rc2 (https://github.com/NLnetLabs/krill?tag=v0.10.0-rc2#269e9fda)`, intermediate artifacts can be found at `/tmp/cargo-install0pBOgH`
```

**Comment by ximon18** [comment]
A workaround is to force the Cryptoki crate that provides the low level PKCS#11 support to generate bindings at compile time rather than using pregenerated bindings (which are missing for the `x86_64-unknown-freebsd` Rust target):

```
cargo install --locked --git https://github.com/NLnetLabs/krill --tag v0.10.0-rc2 --features cryptoki/generate-bindings
```

I have tested the resulting build of Krill on FreeBSD 13.1 with SoftHSM2 using the steps defined [here](https://github.com/NLnetLabs/krill/issues/553) and everything worked.

**Comment by ximon18** [comment]
I have raised https://github.com/parallaxsecond/rust-cryptoki/pull/94 to add support for FreeBSD to the Cryptoki library "out-of-the-box". Once that is approved, merged and included in a new release we can configure Krill to build with the new Cryptoki release and this problem should then be resolved.

**Comment by ximon18** [comment]
Support for FreeBSD was delivered with release 0.1.4 of the [cryptoki-sys crate](https://github.com/parallaxsecond/rust-cryptoki/tree/cryptoki-sys-0.1.4) ([CHANGELOG](https://github.com/parallaxsecond/rust-cryptoki/blob/cryptoki-sys-0.1.4/CHANGELOG.md#cryptoki-sys-014-2022-08-03)) and included in the Kril v0.10.0-rc3 and final [v0.10.0](https://github.com/NLnetLabs/krill/releases/tag/v0.10.0) releases.
